contracts/TokenRequest.sol

```diff
@@ -1,161 +1,157 @@
-    string private constant ERROR_DEPOSIT_NOT_ACTIVE = "TOKENREQUEST_DEPOSIT_NOT_ACTIVE";
-    string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKENREQUEST_TOKEN_TRANSFER_REVERT";
-    string private constant ERROR_TOKEN_APPROVE_FAILED = "TOKENREQUEST_TKN_APPROVE_FAILED";
-
-     struct Request {
+
+    using SafeERC20 for ERC20;
+    using UintArrayLib for uint256[];
+
+    bytes32 constant public SET_TOKEN_MANAGER_ROLE = keccak256("SET_TOKEN_MANAGER_ROLE");
+    bytes32 constant public SET_VAULT_ROLE = keccak256("SET_VAULT_ROLE");
+    bytes32 constant public FINALISE_TOKEN_REQUEST_ROLE = keccak256("FINALISE_TOKEN_REQUEST_ROLE");
+
+    string private constant ERROR_NO_AMOUNT = "TOKEN_REQUEST_NO_AMOUNT";
+    string private constant ERROR_NO_DEPOSIT = "TOKEN_REQUEST_NO_DEPOSIT";
+    string private constant ERROR_ETH_VALUE_MISMATCH = "TOKEN_REQUEST_ETH_VALUE_MISMATCH";
+    string private constant ERROR_TOKEN_TRANSFER_REVERTED = "TOKEN_REQUEST_TOKEN_TRANSFER_REVERTED";
+
+    struct TokenRequest {
+        address requesterAddress;
+        address depositToken;
         uint256 depositAmount;
-        address token;
-        uint256 mintAmount;
-        bool active;
+        uint256 requestAmount;
     }
 
-    Vault public vault;
     TokenManager public tokenManager;
-    Voting public voting;
-    MiniMeToken private token;              //temporary workaround, to show amount of tokens on radspecs's redeem function
-
-    // In case of refund how can we know which pending refund to do?
-    mapping (address => mapping (uint256 => Request)) internal requests;
-    mapping (address => uint256) public requestsLengths;
-
-    event Request(address indexed receiver, address depositToken, uint256 depositAmount, uint256 mintAmount, uint256 voteId, uint256 requestId);
-    event ApprovedRequest(
-        address indexed receiver,
-        address depositToken,
-        uint256 depositAmount,
-        uint256 mintAmount,
-        uint256 voteId,
-        uint256 requestId
-        );
-
-     /**
-    * @notice Initialize TokenRequest app contract
-    * @param _vault Address of the vault
-    * @param _tokenManager TokenManager address
-    * @param _voting Address of the voting
-    */
-    function initialize(Vault _vault, TokenManager _tokenManager, Voting _voting) external onlyInit {
+    address public vault;
+
+    uint256 public nextTokenRequestId;
+    mapping(uint256 => TokenRequest) public tokenRequests; // ID => TokenRequest
+    mapping(address => uint256[]) public addressesTokenRequestIds; // Sender address => List of ID's
+
+    event TokenRequestCreated(uint256 requestId);
+    event TokenRequestRefunded(address refundToAddress, address refundToken, uint256 refundAmount);
+    event TokenRequestFinalised(address requester, address depositToken, uint256 depositAmount, uint256 requestAmount);
+
+    function initialize(address _tokenManager, address _vault) public onlyInit {
```

Comment:
Do you mean why is it `public` instead of `external`? If that's your question personally I don't think it matters. We can change it for consistency though.

Reply:
My question was more related with why we initialize with `initialize(address _tokenManager, address _vault)` in the TokenRequest. But we use `initialize(Vault _vault, TokenManager _tokenManager)` in the Redemptions.

But as you answer in the previous message, it's much clear now that we want to generalize the Vault as much as possible.

Reply:
In Redemptions we want to transfer out of the Vault so we need to be able to call the `transfer` function on it and therefore the vault address must be cast as type Vault.